common/appointment/create.ts

```diff
@@ -0,0 +1,131 @@
+            duration: appointmentToBeCreated.duration,
+            matchId: appointmentToBeCreated.matchId,
+            appointmentType: lecture_appointmenttype_enum.match,
+            organizerIds: [studentUserId],
+            participantIds: [pupilUserId],
+        },
+    });
+};
+
+export const createMatchAppointments = async (matchId: number, appointmentsToBeCreated: AppointmentCreateMatchInput[]) => {
+    const match = await prisma.match.findUniqueOrThrow({ where: { id: matchId } });
+    const student = await prisma.student.findUniqueOrThrow({ where: { id: match.studentId } });
+    const pupil = await prisma.pupil.findUniqueOrThrow({ where: { id: match.pupilId } });
+    const studentUserId = getUserIdTypeORM(student);
+    const pupilUserId = getUserIdTypeORM(pupil);
+    return await Promise.all(
+        appointmentsToBeCreated.map(
+            async (appointmentToBeCreated) =>
+                await prisma.lecture.create({
+                    data: {
+                        title: appointmentToBeCreated.title,
+                        description: appointmentToBeCreated.description,
+                        start: appointmentToBeCreated.start,
+                        duration: appointmentToBeCreated.duration,
+                        matchId: appointmentToBeCreated.matchId,
+                        appointmentType: lecture_appointmenttype_enum.match,
+                        organizerIds: [studentUserId],
+                        participantIds: [pupilUserId],
+                    },
+                })
+        )
+    );
+};
+
+export const createGroupAppointment = async (appointmentToBeCreated: AppointmentCreateGroupInput) => {
+    const participants = await prisma.subcourse_participants_pupil.findMany({
+        where: { subcourseId: appointmentToBeCreated.subcourseId },
+        select: { pupil: true },
+    });
+    const instructors = await prisma.subcourse_instructors_student.findMany({
+        where: { subcourseId: appointmentToBeCreated.subcourseId },
+        select: { student: true },
+    });
+    assert(
+        instructors.length > 0,
+        `No instructors found for Subcourse(${appointmentToBeCreated.subcourseId}) there must be at least one organizer for an appointment`
+    );
+    return await prisma.lecture.create({
+        data: {
+            title: appointmentToBeCreated.title,
+            description: appointmentToBeCreated.description,
+            start: appointmentToBeCreated.start,
+            duration: appointmentToBeCreated.duration,
+            subcourseId: appointmentToBeCreated.subcourseId,
+            appointmentType: lecture_appointmenttype_enum.group,
+            organizerIds: instructors.map((i) => getUserForTypeORM(i.student).userID),
+            participantIds: participants.map((p) => getUserForTypeORM(p.pupil).userID),
+        },
+    });
+};
+
+export const createGroupAppointments = async (subcourseId: number, appointmentsToBeCreated: AppointmentCreateGroupInput[]) => {
```

Comment:
What's the purpose of having two different functions? Couldn't we use `createGroupAppointments`  all the time?

Reply:
> I guess from the UX perspective it's better to have different endpoints for different use cases. It is planned to send different notifications if a single appointment or more appointments are created.

Same here